### PR TITLE
Support @Trace, @Tag and @Tags work for static methods

### LINKED ISSUE #6640: apm-toolkit-trace: @Trace, @Tag and @Tags do not work for static methods
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [x] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
I added `Trace` annotation to a static method. The result shows that `Trace` annotation does not work for static methods. A browse of `apm-toolkit-trace-activiation` module shows that `TagAnnotationActivation` and `TraceAnnotationActivation` only do method interception for static methods.

Does this work as designed? If not, I am willing to add support for static methods.

### PR DESCRIPTION
**Pull request by liqiangz**
### Fix <bug description or the bug issue number or bug issue link>
- [x] Add a unit test to verify that the fix works.
- [x] Explain briefly why the bug exists and how to fix it.
@Trace, @Tag and @Tags do not work for static methods. Add StaticMethodsAroundInterceptor to support static methods.

- [x] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #6640.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Hi, thanks for submitting. I think there is one decision to be made. Today, we don't support to use `this.field` as tag value, so, trace/tag instance and static methods look the same. But consider those cases, they are not.
What do you think about separating these 2 cases, and we may need to add support for `this.field` one day.

**Comment by wu-sheng** [comment]
Reference doc, http://skywalking.apache.org/docs/main/v8.4.0/en/setup/service-agent/java-agent/application-toolkit-trace/. We only support using parameter and returnObj as tag value.

**Comment by liqiangz** [comment]
@wu-sheng  In most cases, these two annotations are the same. I feel separating the two cases for `this.filed`  increases the cost of use. If we need to support `this.field`  one day, maybe we can simply set tag value to `null` for static methods.

**Comment by wu-sheng** [comment]
> I feel separating the two cases for this.filed will increase the cost of use

OK, make sense to me. Let's focus on the details.

File: apm-sniffer/apm-toolkit-activation/apm-toolkit-trace-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/trace/TagStaticAnnotationActivation.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import static org.apache.skywalking.apm.agent.core.plugin.match.logical.LogicalMatchOperation.not;
+import static org.apache.skywalking.apm.agent.core.plugin.match.logical.LogicalMatchOperation.or;
+import static org.apache.skywalking.apm.toolkit.activation.trace.TagAnnotationActivation.TAGS_ANNOTATION;
+import static org.apache.skywalking.apm.toolkit.activation.trace.TagAnnotationActivation.TAG_ANNOTATION;
+import static org.apache.skywalking.apm.toolkit.activation.trace.TraceAnnotationActivation.TRACE_ANNOTATION;
+
+/**
+ * The static of tag annotation activation.
+ */
+public class TagStaticAnnotationActivation extends ClassStaticMethodsEnhancePluginDefine {
```
Comment (wu-sheng) [inline apm-sniffer/apm-toolkit-activation/apm-toolkit-trace-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/trace/TagStaticAnnotationActivation.java:40]: You don't need a new activation, you could enhance the existing one by `extends ClassEnhancePluginDefine`.

Comment (liqiangz) [reply apm-sniffer/apm-toolkit-activation/apm-toolkit-trace-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/trace/TagStaticAnnotationActivation.java]: done

**Comment by wu-sheng** [comment]
FYI @yaojingguo

**Review by wu-sheng — APPROVED** [review]
LGTM. Wait for @ascrutae to recheck.

**Review by ascrutae — APPROVED** [review]

**Pull request merged**